src/main/java/Duke.java

```diff
@@ -1,10 +1,159 @@
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
 public class Duke {
+    public static String h_rule = "────────────────────────────────────────────────────────────────────";
+
+    public static int dukeRunner(String log) throws DukeNotFoundException, DukeDescriptionException, DukeTimingException, FileNotFoundException {
+        Scanner sc = new Scanner(System.in);
+
+        String raw_in;
+        String[] input;
+        List<Task> itemList = new ArrayList<>();
+        int count = 0;
+        List<String> fileContents = null;
+
+        try {
+            fileContents = new ArrayList<>(Files.readAllLines(Paths.get(log)));
+        } catch (IOException e) {
+            System.out.println(e);
+        }
+        while (true) {
+            raw_in = sc.nextLine().trim();
+            input = raw_in.split(" ");
+            System.out.println(h_rule);
+            String joined = "";
+            String timing = "";
+            if (raw_in.equals("bye")) {
+                System.out.println("Goodbye! Have a nice day!\n");
+                break;
+            } else {
+                switch (input[0]) {
+                    case "list":
+                        int index = 1;
+                        for (Task s : itemList) {
+                            System.out.println(index + ". " + s.toString());
+                            index++;
+                        }
+                        break;
+                    case "done":
+                        int itemToBeUpdatedIndex = Integer.parseInt(input[1]) - 1;
+                        Task task = itemList.get(itemToBeUpdatedIndex);
+                        task.markDone();
+                        fileContents.set(itemToBeUpdatedIndex, task.toString());
+                        System.out.println("Alright, I will mark this as done.\n" + input[1] + ". " + task.toString());
+                        break;
+                    case "todo":
+                        for (int i = 1; i < input.length; i++) {
```

Comment:
This might be able to be abstracted out using the `String.join` function.

This kind of abstraction may be applicable in a few other parts of the code base too.